src/timer/TimerProvider.jsx

```diff
@@ -77,26 +77,34 @@ const TimerServiceProvider = ({
   };
 
   useEffect(() => {
-    let secondsLeft = startValue;
     let timerTick = 0;
-    const interval = setInterval(() => {
-      secondsLeft -= 1;
-      timerTick += 1;
-      setTimeState(getFormattedRemainingTime(secondsLeft));
-      processTimeLeft(interval, secondsLeft);
-      // no polling during grace period
-      if (timerTick % POLL_INTERVAL === 0 && secondsLeft >= 0) {
-        pollExam();
-      }
-
-      // if exam is proctored ping provider app also
-      if (workerUrl && timerTick % pingInterval === pingInterval / 2) {
-        pingHandler(pingInterval, workerUrl);
+    if (liveInterval) {
+      clearInterval(liveInterval);
+      liveInterval = null;
+    } else {
```

Comment:
can you explain why this is in an else statement? Don't we want to create a new timer after clearing the old one?

Reply:
You are right. That's a mistake I made here. Thank you for pointing that out. Will update

Reply:
Updated. @Zacharis278 Please review again.